lnrpc/walletrpc/walletkit.proto

```diff
@@ -0,0 +1,157 @@
+}
+
+message AddrRequest{
+    // No fields, as we always give out a p2wkh address.
+}
+message AddrResponse {
+    /**
+    The address encoded using a bech32 format.
+    */
+    string addr = 1;
+}
+
+message Transaction {
+    /**
+    The raw serialized transaction.
+    */
+    bytes tx_hex = 1;
+}
+message PublishResponse {
+    /**
+    If blank, then no error occurred and the transaction was successfully
+    published. If not the empty string, then a string representation of the
+    broadcast error.
+
+    TODO(roasbeef): map to a proper enum type
+    */
+    string publish_error = 1;
+}
+
+message SpendReport {
+    /**
+    If true, then this outpoint has been spent a confirmed or unconfirmed
+    transaction within the internal transaction store of the wallet.
+    */
+    bool has_spend = 1;
+
+    /**
+    The raw transaction that spends the target outpoint. Note that this will
+    only be populated if the above field returns true.
+    */
+    bytes raw_tx_hex = 2;
+}
+
+message OutPoint {
+    /**
+    The txid of the transaction that produced this output.
+    */
+    bytes txid = 1;
+
+    /**
+    The index of the output within the transaction above that is being
+    referenced.
+    */
+    int32 index = 2;
+}
+
+message SendOutputsRequest {
+    /**
+    The number of satoshis per kilo weight that should be used when crafting
+    this transaction.
+    */
+    int64 sat_per_kw = 1;
+
+    /**
+    A slice of the outputs that should be created in the transaction produced.
+    */
+    repeated signrpc.TxOut outputs = 2;
+}
+message SendOutputsResponse {
+    /**
+    The txid of the transaction sent out on the network.
+    */
+    bytes txid = 1;
+}
+
+message EstimateFeeRequest {
+    /**
+    The number of confirmations to shoot for when estimating the fee.
+    */
+    int32 conf_target = 1;
+}
+message EstimateFeeResponse {
+    /**
+    The amount of satoshis per kw that should be used in order to reach the
+    confirmation target in the request.
+    */
+    int64 sat_per_kw = 2;
+}
+
+service WalletKit {
+    /**
+    DeriveNextKey attempts to derive the *next* key within the key family
+    (account in BIP43) specified. This method should return the next external
+    child within this branch.
+    */
+    rpc DeriveNextKey(KeyReq) returns (signrpc.KeyDescriptor);
+
+    /**
+    DeriveKey attempts to derive an arbitrary key specified by the passed
+    KeyLocator. 
+    */
+    rpc DeriveKey(signrpc.KeyLocator) returns (signrpc.KeyDescriptor);
+
+    /**
+    NextAddr returns the next unused address within the wallet.
+    */
+    rpc NextAddr(AddrRequest) returns (AddrResponse);
+
+    /**
+    PublishTransaction attempts to publish the passed transaction to the
+    network. Once this returns without an error, the wallet will continually
+    attempt to re-broadcast the transaction on start up, until it enters the
+    chain.
+    */
+    rpc PublishTransaction(Transaction) returns (PublishResponse);
```

Comment:
How is this going to work with the rebroadcaster? It needs sign descriptors too.

Reply:
Suggestion: instead of `PublishTransaction`, this could be `SendInputs` which takes sign descriptors.

Reply:
So just to check in again, I think we're find with `PublishTransaction` and may follow up with the two-stage craft+sign process?